Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,230 @@
+                    //        }]
+                    //    }
+                    //};
+                    //var limitation = { // DUMMY SUBTREE DATA
+                    //    "_identifier": "Subtree",
+                    //    "values": {
+                    //        "ref": [{
+                    //            "_href": "/api/ezp/v2/content/locations/1/5/14",
+                    //            "_media-type": "application/vnd.ez.api.Location+json"
+                    //        }]
+                    //    }
+                    //};
+
+                    // Convert hrefs to IDs, required by API
+                    if (limitation) {
+                        for ( var i = 0; i < limitation.values.ref.length; i++ ) {
+                            if (limitation._identifier == "Section") {
+                                limitation.values.ref[i]._href = /\d+$/.exec(limitation.values.ref[i]._href)[0];
+                            } else if (limitation._identifier == "Subtree") {
+                                limitation.values.ref[i]._href = /(\/\d+)+$/.exec(limitation.values.ref[i]._href)[0];
+                            }
+                        }
+                    }
+
+                    var roleAssignInputStruct = userService.newRoleAssignInputStruct(role, limitation),
+                        contentTypeIdentifier = struct.contentType.get('identifier');
+
+                    if (contentTypeIdentifier === 'user') {
+                        discoveryService.getInfoObject('users', function (error, usersInfoObject) {
+                            if (error) {
+                                service._error('getInfoObject for "users" failed: ' + error.message);
+                                return;
+                            }
+
+                            // We need a user ID, e.g.: /api/ezp/v2/user/users/10
+                            var userIdStr = usersInfoObject._href + '/' + struct.content.get('contentId');
+
+                            userService.assignRoleToUser(
+                                userIdStr, roleAssignInputStruct, end
+                            );
+                        });
+                    } else if (contentTypeIdentifier === 'user_group') {
+                        discoveryService.getInfoObject('rootUserGroup', function (error, rootUserGroup) {
+                            if (error) {
+                                service._error('getInfoObject for "rootUserGroup" failed: ' + error.message);
+                                return;
+                            }
+
+                            /* We need to convert content ID to user group ID, e.g.
+                               change this: /api/ezp/v2/content/locations/1/5/14
+                               into this: /api/ezp/v2/user/groups/1/5/14
+                             */
+                            var location = struct.content.get('resources').MainLocation;
```

Comment:
That and the next line: the same as https://github.com/ezsystems/PlatformUIBundle/pull/344/files#r41520575 (declaration of variable in the beginning of function)